lib/grape/error_formatter/base.rb

```diff
@@ -3,9 +3,13 @@ module ErrorFormatter
     module Base
       def present(message, env)
         present_options = {}
-        present_options[:with] = message.delete(:with) if message.is_a?(Hash)
+        presented_message = message
+        if presented_message.is_a?(Hash) && presented_message.key?(:with)
+          present_options[:with] = presented_message[:with]
+          presented_message = presented_message.reject { |k, _| k == :with }
+        end
```

Comment:
Seems a little convoluted, why don't you just `message.dup`? It's probably a good idea anyway because users may want to use the response from `present` and modify it further.